cypress/integration/device_page_test.js

```diff
@@ -0,0 +1,27 @@
+describe('Device page tests', () => {
+  context('no access before login', () => {
+    before(() => {
+      cy.fixture('device').as('device');
+    });
+
+    it('redirects to home', function () {
+      cy.visit(`/devices/${this.device.data.id}`);
+      cy.location('pathname').should('eq', '/login');
+    });
+  });
+
+  context('authenticated', () => {
+    beforeEach(() => {
+      cy.fixture('device').as('device');
+      cy.server();
+      cy.route('GET', '/appengine/v1/*/devices/*', '@device');
+    });
+
+    it('successfully loads', function () {
+      cy.login();
+      cy.visit(`/devices/${this.device.data.id}`);
+      cy.location('pathname').should('eq', `/devices/${this.device.data.id}`);
+      cy.get('h2').contains('Device');
```

Comment:
we should at least test more device status information, such as connected, last_connection, last_disconnection etc...

Reply:
bonus point: testing aliases, metadata and groups

Reply:
Nice advice. I added more checks on device properties